src/charm.py

```diff
@@ -454,7 +448,7 @@ def _on_vault_pki_certificate_creation_request(
     def _generate_pki_certificate_for_requirer(self, csr: str, relation_id: int):
         """Generate a PKI certificate for a TLS requirer."""
         if not self.unit.is_leader():
-            logger.debug("Only leader unit can handle a vault-pki certificate request")
+            logger.debug("Only leader unit can handle a vault-pki certificate request")  # RFC: Why
```

Comment:
Because we want to avoid TLS provider units competing to provide a cert, the cert is provided by the Vault app, not unit. Plus only the leader unit can write to the application relation data.